versionedRouter.js

```diff
@@ -878,10 +879,11 @@ async function handleProxyRequest(destination, ctx) {
     return {};
   };
 
-  const destinationRequest = ctx.request.body;
+  const { metadata, destinationRequest } = ctx.request.body;
```

Comment:
This would be a breaking change. We need to make these changes backward compatible with previous server versions.
CC @sanpj2292

Reply:
@ujjwal-ab Let's send the metadata information as part of the destinationRequest itself, so that the backward compatibility is maintained